src/python_gardenlinux_lib/cname.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+from .features import parse_features
+
+from functools import reduce
+from os.path import basename, dirname
+
+import argparse
+import re
+
+
+def main():
```

Comment:
Just for your reference, we also have the [python-gardenlinux-cli](https://github.com/gardenlinux/python-gardenlinux-cli), which is used as a CLI wrapper for this library.

Reply:
... at least this was the initial intention. :D 

not sure how to design this now, since [python-gardenlinux-cli](https://github.com/gardenlinux/python-gardenlinux-cli) looks more like a tool for OCI only.

Reply:
I'm not sure if adding a separate CLI repository may not be counter productive as reusability of code between "lib" and "cli" parts are _at least_ more complicated in that case. Naming of both repositories may be changed to reflect there content.

Furthermore GardenLinux does not have a Python code base that allows "lib" code to be used without a "cli" implementation. Therefore I would suggest to unify the code paths in place.